### PR TITLE
Update docs for pulling toggle state data over HTTP; some cleanup

### PR DESCRIPTION
**Pull request by timmc-edx**
Including:

- Preliminary annotations integration for credentials
- Various typo and doc syntax fixes
- Remove gather_feature_toggle_state.py which is no longer used
  (now getting state via HTTP instead of direct DB access);
  accordingly, remove MySQL dependency
- Delete references to devstack make target for gathering toggles
  (deleted in https://github.com/edx/devstack/pull/648)
- Remove reference to `rename` option (apparently deleted in 56ec624638e)

### CODE REVIEW

File: docs/how_to/create_toggles_report_for_devstack_or_sandbox.rst [hunk 1]
```diff
@@ -25,8 +25,7 @@ First create this empty directory. Now populate it with data based on either of
 When creating report to compare multiple environments
 +++++++++++++++++++++++++++++++++++++++++++++++++++++
 
-For each env, create a directory named following this name pattern: `<env_name>_env`.
-Populate these env directories with their state value files (json).
+For each env, create a directory named following this name pattern: `<env_name>_env`. Populate these env directories with their state value files (json).
```
Comment (robrap) [inline docs/how_to/create_toggles_report_for_devstack_or_sandbox.rst:28]: This, and other sections, feel like they belong elsewhere:
1. This doc claims to be "How-to: Enabling toggles report for a new IDA", which should be a list of steps needed to accomplish this.  Maybe the doc I want is to edx.org specific?
2. The "Problems with current approach" feel like they belong in code comments or maybe an ADR about some implementation, etc.  Maybe small parts of it belong in this file as a warning?
3. I don't know where this section belongs?  This seems like more manual instructions on how to gather data for generating the report.  This seems important, like for devstack, sandboxes, Open edX, etc., but this feels like the wrong how-to.

File: docs/how_to/create_toggles_report_for_devstack_or_sandbox.rst [hunk 2]
```diff
@@ -50,8 +49,8 @@ Example:
         - discovery_waffle.json
 
 
-Grab annoatations data
-----------------------
+Grab annotatations data
```
Comment (robrap) [inline docs/how_to/create_toggles_report_for_devstack_or_sandbox.rst:52]: Too many tatas. :)

**Review by robrap — COMMENTED** [review]
@timmc-edx: Some minor comments, and then a comment about what belongs where that I am happy to discuss synchronously if you want.

File: docs/how_to/create_toggles_report_for_devstack_or_sandbox.rst (hunk 1 above)
Comment (timmc-edx) [reply docs/how_to/create_toggles_report_for_devstack_or_sandbox.rst:28]: Agreed. I really struggled with how to refactor this doc, and the others -- the information is spread out across about 4 repositories and some combination of duplicated, out of date, and not properly stratified into Open edX/edx.org. I'll put something on your schedule.

Comment (timmc-edx) [reply docs/how_to/create_toggles_report_for_devstack_or_sandbox.rst:28]: Sounds about right.

File: docs/how_to/adding_new_ida_to_toggle_report.rst [hunk 3]
```diff
@@ -56,56 +58,63 @@ steps
...
-    - it will block collection of annotation in other repositories
-    - archbom is alway notified on failure
-    - fix: Use the same approach as the upgrade jobs, create multiple jobs based on list which contains repository link, and contact email. One job per IDA/repository
-* need to add custom script to every IDA that needs to be in IDA repository
-    - fix: create shell script in job to do the job instead. code_annotations should be able to work across repositories and any custom commands should not be necessary
+- If job fails:
+
+    - It will block collection of annotation in other repositories
+    - Arch-bom is alway notified on failure
+    - Fix: Use the same approach as the upgrade jobs, create multiple jobs based on list which contains repository link, and contact email. One job per IDA/repository.
```
[the remark this answers is not in the document]
Comment (timmc-edx) [reply docs/how_to/adding_new_ida_to_toggle_report.rst]: I was able to delete most of this text now that this is generalized.

File: docs/how_to/adding_new_ida_to_toggle_report.rst [hunk 4]
```diff
@@ -56,56 +58,63 @@ steps
...
 
-* if job fails
-    - it will block collection of annotation in other repositories
-    - archbom is alway notified on failure
-    - fix: Use the same approach as the upgrade jobs, create multiple jobs based on list which contains repository link, and contact email. One job per IDA/repository
-* need to add custom script to every IDA that needs to be in IDA repository
-    - fix: create shell script in job to do the job instead. code_annotations should be able to work across repositories and any custom commands should not be necessary
+- If job fails:
+
+    - It will block collection of annotation in other repositories
```
[the remark this answers is not in the document]
Comment (timmc-edx) [reply docs/how_to/adding_new_ida_to_toggle_report.rst]: I think the indentation was too deep -- perhaps muscle memory from Markdown.

File: scripts/README.rst [hunk 5]
```diff
@@ -1,68 +1,31 @@
...
-an application's database.
+In order to create a feature toggle report for a given deployment, you need two types of data: feature toggle annotation data and feature toggle state data from an application's database.
 
 Feature Toggle Annotation Data
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Use instructions found in `get feature toggles annotation data document <https://edx-toggles.readthedocs.io/en/latest/how_to/documenting_new_feature_toggles.html>`__ to collect annotations data.
+Any toggles that are annotated according to the instructions in
+`get feature toggles annotation data document <docs/how_to/documenting_new_feature_toggles.html>`_
+will be included in the report.
```
Comment (robrap) [inline scripts/README.rst:15]:
```suggestion
Any toggles that are annotated according to the instructions in `get feature toggles annotation data document <docs/how_to/documenting_new_feature_toggles.html>`_ will be included in the report.
```

**Review by robrap — APPROVED** [review]
I've got one comment fix.  Thanks!

File: scripts/README.rst (hunk 5 above)
Comment (timmc-edx) [reply scripts/README.rst]: Hah! Thanks. It's very hard for me to spot these.

**Pull request merged**